### PR TITLE
CY-693 : adding protection against deleting imported blueprints

### PR DESCRIPTION
**Pull request by tyacbovi**
- The protection will be against deleting blueprints in use in another blueprint (even not in live deployments).
- The force delete option was added in case of need
- Merging two unit tests files that are on the same topic

### CODE REVIEW

File: rest-service/dev-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/cloudify-cosmo/cloudify-common@master#egg=cloudify-common==4.5.5.dev1
+git+https://github.com/cloudify-cosmo/cloudify-common@CY-693#egg=cloudify-common==4.5.5.dev1
```
Comment (geokala) [inline rest-service/dev-requirements.txt]: This shouldn't be in a PR

File: rest-service/manager_rest/manager_exceptions.py [hunk 2]
```diff
@@ -429,6 +429,18 @@ def __init__(self, *args, **kwargs):
         )
 
 
+class BlueprintInUseError(ManagerException):
```
Comment (geokala) [inline rest-service/manager_rest/manager_exceptions.py]: Don't we already have something like this error when you try to delete a blueprint that an existing deployment uses?

File: rest-service/manager_rest/rest/resources_v3_1/blueprints.py [hunk 3]
```diff
@@ -72,3 +75,23 @@ def put(self, blueprint_id, **kwargs):
         return UploadedBlueprintsManager().\
             receive_uploaded_data(data_id=blueprint_id,
                                   visibility=visibility)
+
+    @swagger.operation(
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v3_1/blueprints.py]: I didn't realise we were missing this in swagger before. Oh dear. We should probably have some tests checking that all of our API paths are actually in swagger?

Comment (geokala) [reply rest-service/manager_rest/rest/resources_v3_1/blueprints.py]: (but not as part of this)

Comment (tyacbovi) [reply rest-service/manager_rest/rest/resources_v3_1/blueprints.py]: Agree, also there should be work on updating the swagger docs and fixing our swagger specification generation

File: rest-service/dev-requirements.txt (hunk 1 above)
Comment (tyacbovi) [reply rest-service/dev-requirements.txt]: Need here so the unit tests can be run, because this change depends on common change

File: rest-service/manager_rest/manager_exceptions.py (hunk 2 above)
Comment (tyacbovi) [reply rest-service/manager_rest/manager_exceptions.py]: Nope, there isn't

File: rest-service/manager_rest/resource_manager.py [hunk 4]
```diff
@@ -379,17 +379,43 @@ def publish_blueprint(self,
...
+                    'blueprint removal.'.format(blueprint_id))
+
         if len(blueprint.deployments) > 0:
             raise manager_exceptions.DependentExistsError(
                 "Can't delete blueprint {0} - There exist "
                 "deployments for this blueprint; Deployments ids: {1}"
                 .format(blueprint_id,
                         ','.join([dep.id for dep
                                   in blueprint.deployments])))
+        # Delete blueprint resources from file server
```
Comment (mcouthon) [inline rest-service/manager_rest/resource_manager.py]: Why did you move it here?

Comment (tyacbovi) [reply rest-service/manager_rest/resource_manager.py]: So I should not need to copy this in the V3.1 and that's the more relevant place (like the plugins)

File: rest-service/manager_rest/test/endpoints/test_blueprints.py [hunk 5]
```diff
@@ -36,18 +37,18 @@ def test_get_empty(self):
     def test_get_nonexistent_blueprint(self):
         try:
             self.client.blueprints.get('15')
-        except CloudifyClientError, e:
+        except exceptions.CloudifyClientError, e:
```
Comment (tehasdf) [inline rest-service/manager_rest/test/endpoints/test_blueprints.py]: if we're already changing those lines, please also change the `except E, e` to `except E as e`, which because the comma syntax is old and deprecated (and stops working in py3)

File: rest-service/manager_rest/test/endpoints/test_blueprints.py [hunk 6]
```diff
@@ -130,8 +131,9 @@ def test_delete_blueprint(self):
                 post_blueprints_response['id'], 'blueprint.yaml'))
 
         # trying to delete a nonexistent blueprint
-        resp = self.delete('/blueprints/nonexistent-blueprint')
-        self.assertEquals(404, resp.status_code)
+        with self.assertRaises(exceptions.CloudifyClientError) as context:
```
Comment (tehasdf) [inline rest-service/manager_rest/test/endpoints/test_blueprints.py]: we can change that to a specific exception type (not the catch-all CloudifyClientError), and then we won't need to assert the 404
